internal/model/puller.go

```diff
@@ -327,7 +329,17 @@ func (p *Puller) pullerIteration(ignores *ignore.Matcher) int {
 		switch {
 		case file.IsDeleted():
 			// A deleted file, directory or symlink
-			deletions = append(deletions, file)
+			if file.IsDirectory() {
+				dirDeletions = append(dirDeletions, file)
+			} else {
+				fileDeletions[file.Name] = file
+				df, ok := p.model.CurrentFolderFile(p.folder, file.Name)
+				if ok {
+					// Put files into buckets per first hash
+					key := string(df.Blocks[0].Hash)
```

Comment:
Out of range here, so df has no blocks apparently. Maybe print it at figure out what it is. Already deleted, but then why should we delete it again?